src/DynamoWebServer/Responses/ArrayItemsDataResponse.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using Dynamo.Models;
+using ProtoCore.Mirror;
+
+namespace DynamoWebServer.Responses
+{
+    class ArrayItemsDataResponse: Response
+    {
+        /// <summary>
+        /// Guid of the specified code block node
+        /// </summary>
+        [DataMember]
+        public string NodeId { get; private set; }
+
+        /// <summary>
+        /// String representing of the data about input,
+        /// output ports, text of specified code block node
+        /// </summary>
+        [DataMember]
+        public IEnumerable<string> Items { get; private set; }
+
+        /// <summary>
+        /// Index from which Dynamo responds array items
+        /// </summary>
+        [DataMember]
+        public int IndexFrom { get; set; }
+
+        public ArrayItemsDataResponse(NodeModel node, int indexFrom, int length)
+        {
```

Comment:
`node.CachedValue` gets reset to `null` once in a while during execution (in order to invalidate the previous value), and it becomes briefly unavailable before the value query returns with the latest `MirrorData`. I'm sure we need to check for `null` cases.